pyro/ops/welford.py

```diff
@@ -48,3 +48,46 @@ def get_covariance(self, regularize=True):
                 scaled_cov.view(-1)[::scaled_cov.size(0) + 1] += shrinkage
                 cov = scaled_cov
         return cov
+
+
+class WelfordArrowheadCovariance:
```

Comment:
Would this interface also be able to support arrowhead precision structure, or would we need to replace `.get_covariance()` with a matrix-vector-product interface?

Reply:
Hi @fritzo, I have moved WelfordArrowheadCovariance to #2465. In #2465, `get_covariance` will return the `top` part and `bottom_diag` part of the arrowhead matrix (to save memory).

Reply:
I'm confused. I see this could possibly return a compact representation of an arrowhead covariance matrix, but my question is about arrowhead precision matrices. In contrast to arrowhead covariance matrices, the covariance corresponding to an arrowhead precision matrix is dense. Stated another way, if we implement not only a `WelfordArrowheadCovariance` but also a `WelfordArrowheadPrecision` whose covariance is dense, will this `.get_covariance()` method still be tractable, or will we need to implement a custom matrix-vector product op (which is cheap for arrowhead-precisions; we can solve against a sparse precision matrix).

Reply:
Yes, in #2465, we learn arrowhead precision matrix (i.e. arrowhead mass matrix). The covariance (inverse mass matrix) will be dense. But the sqrt of the covariance will be a sparse triangular matrix. We will leverage this sparse structure to cheaply compute matrix-vector product with O(N x head_size^2) cost.

Reply:
Thanks for explaining!